library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Block.hs

```diff
@@ -646,6 +647,34 @@ printf args = do
   printf <- externf "printf"
   instr Type.i32 $ callConvention CC.C printf (emptyArgs args)
 
+-- | @globalString@ creates a globally-defined string constant and
+-- returns a pointer to it.
+globalString ::
+  ( RetInstruction m,
+    HasState "moduleDefinitions" [Definition] m
+  ) =>
+  [Char] ->
+  Name ->
+  m Operand
+globalString str name = do
+  addDefn $
+    GlobalDefinition $
+      globalVariableDefaults
+        { Global.name = name,
+          Global.initializer = Just (cString str),
+          Global.type' = ArrayType (fromIntegral (length str + 1)) Type.i8
+        }
+  getElementPtr $
+    Types.Minimal
+      { Types.type' = Types.pointerOf Type.i8,
+        Types.address' =
+          ConstantOperand $
+            C.GlobalReference
+              (Types.pointerOf $ ArrayType (fromIntegral (length str + 1)) Type.i8)
+              name,
+        Types.indincies' = constant32List [0, 0]
```

Comment:
```suggestion
        Types.indices' = constant32List [0, 0]
```

Reply:
We could certainly make that change, but that spelling is in pre-existing code, so we can't make it without changing the other code which defines and uses it.  I'd suggest a separate PR for that, as it's a separate matter from supporting strings (a cosmetic one).